packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleValue.tsx

```diff
@@ -113,7 +113,7 @@ GridFilterInputMultipleValue.propTypes = {
     operator: PropTypes.string.isRequired,
     value: PropTypes.any,
   }).isRequired,
-  type: PropTypes.oneOf(['number', 'text']),
+  type: PropTypes.oneOf(['number', 'text', 'date', 'datetime-local']),
```

Comment:
May instead remove this, as per [the issue's suggestion](https://github.com/mui/mui-x/issues/13409#issuecomment-2155269895), that would make this consistent with how the other input field does it, type but no enforcement in the prop types for it.

```suggestion
```